### PR TITLE
DS apiserver: Minor fixes

### PR DESCRIPTION
**Pull request by marefr**
- Return after error in query endpoint rather than continue
- Support query parameters in resource endpoint and include incoming HTTP headers

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM

File: pkg/registry/apis/datasource/sub_resource.go [hunk 1]
```diff
@@ -66,12 +67,21 @@ func (r *subResourceREST) Connect(ctx context.Context, name string, opts runtime
 			return
 		}
 
-		path := req.URL.Path[idx+len("/resource"):]
+		clonedReq := req.Clone(req.Context())
```
Comment (andresmgot) [inline pkg/registry/apis/datasource/sub_resource.go:70]: just curious, why the clone?

Comment (marefr) [reply pkg/registry/apis/datasource/sub_resource.go:70]: I copied from
https://github.com/grafana/grafana/blob/169be9e40170b15b2561d7e60c00be65ff6f82ee/pkg/api/plugin_resource.go#L90-L100

Comment (marefr) [reply pkg/registry/apis/datasource/sub_resource.go:70]: Example if there's some middleware running after so we don't want to change the original request

**Pull request merged**